Sources/System/FilePath/FilePathComponents.swift

```diff
@@ -166,15 +176,22 @@ extension _StrSlice {
   internal var _systemString: SystemString { SystemString(_slice) }
 }
 extension _StrSlice {
-  public static func == (lhs: Self, rhs: Self) -> Bool {
+  public static func strSliceEqual(lhs: Self, rhs: Self) -> Bool {
     lhs._slice.elementsEqual(rhs._slice)
   }
-  public func hash(into hasher: inout Hasher) {
+  public static func == (lhs: Self, rhs: Self) -> Bool {
```

Comment:
We should remove this function altogether. (After verifying that it doesn't end up getting exported somehow in the ABI stable binary.)